checkov/sca_package_2/runner.py

```diff
@@ -0,0 +1,151 @@
+        if scan_results is None:
+            return report
+
+        for result in scan_results:
+            if not result:
+                continue
+            package_file_path = Path(result["repository"])
+            if self._code_repo_path:
+                try:
+                    package_file_path = package_file_path.relative_to(self._code_repo_path)
+                except ValueError:
+                    # Path.is_relative_to() was implemented in Python 3.9
+                    pass
+
+            vulnerabilities = result.get("vulnerabilities") or []
+            packages = result.get("packages") or []
+
+            license_statuses = [_LicenseStatus(package_name=elm["packageName"], package_version=elm["packageVersion"],
+                                               policy=elm["policy"], license=elm["license"], status=elm["status"])
+                                for elm in result.get("license_statuses") or []]
+
+            rootless_file_path = str(package_file_path).replace(package_file_path.anchor, "", 1)
+            add_to_report_sca_data(
+                report=report,
+                check_class=self._check_class,
+                scanned_file_path=str(package_file_path),
+                rootless_file_path=rootless_file_path,
+                runner_filter=runner_filter,
+                vulnerabilities=vulnerabilities,
+                packages=packages,
+                license_statuses=license_statuses,
+                report_type=self.report_type,
+            )
+
+        return report
+
+    def upload_scannable_files(
+            self,
+            root_path: Path | None,
+            files: list[str] | None,
+            excluded_paths: set[str],
+            excluded_file_names: set[str] | None = None,
+            root_folder: str | Path | None = ""
+    ) -> bool:
+        """ upload scannable files to s3"""
+        excluded_file_names = excluded_file_names or set()
+        package_files_to_persist: List[FileToPersist] = []
+        if root_path:
+            for file_path in root_path.glob("**/*"):
+                if file_path.name in SUPPORTED_PACKAGE_FILES and not any(
+                        p in file_path.parts for p in excluded_paths) and file_path.name not in excluded_file_names:
+                    file_path_str = str(file_path)
+                    package_files_to_persist.append(
+                        FileToPersist(file_path_str, os.path.relpath(file_path_str, root_folder)))
+
+        for file in files or []:
+            file_path = Path(file)
+            if not file_path.exists():
+                logging.warning(f"File {file_path} doesn't exist")
+                continue
+            package_files_to_persist.append(FileToPersist(file, os.path.relpath(file, root_folder)))
+
+        logging.info(f"{len(package_files_to_persist)} sca package files found.")
+        bc_integration.persist_files(package_files_to_persist)
+        return bool(package_files_to_persist)
```

Comment:
```suggestion
        return package_files_to_persist
```
how about this and then change the type hint to
```python
->  list[FileToPersist]:
```
your if condition where you call the function will still work, because it will check if it empty or not.